Include/internal/pycore_format.h

```diff
@@ -0,0 +1,27 @@
+#ifndef Py_INTERNAL_FORMAT_H
+#define Py_INTERNAL_FORMAT_H
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
I'm fine with adding pycore_format.h.

But maybe we could rename pycore_bytes_methods.h to pycore_bytes.h or pycore_bytesobject.h and put everything in a single header file. It's up to you.

It seems like unicode doesn't use these constants, so maybe be more explicit in the file name and rename it to "pycore_bytesformat.h" or "pycore_bytes_format.h"?

Reply:
These constants are used both by bytes and unicode formatters. This is the reason why they are declared in the header file.

`pycore_bytes_methods.h` contains declarations for `bytes_methods.c` which contains common code and docstrings for bytes and bytearray.